Mathlib/Analysis/Matrix/Order.lean

```diff
@@ -0,0 +1,260 @@
+@[deprecated CFC.sqrt (since := "2025-09-22")]
+noncomputable def sqrt : Matrix n n 𝕜 :=
+  hA.1.eigenvectorUnitary.1 * diagonal ((↑) ∘ (√·) ∘ hA.1.eigenvalues) *
+  (star hA.1.eigenvectorUnitary : Matrix n n 𝕜)
+
+@[deprecated CFC.sqrt_nonneg (since := "2025-09-22")]
+lemma posSemidef_sqrt : PosSemidef (CFC.sqrt A) := CFC.sqrt_nonneg A |>.posSemidef
+
+include hA
+
+@[deprecated CFC.sq_sqrt (since := "2025-09-22")]
+lemma sq_sqrt : (CFC.sqrt A) ^ 2 = A := CFC.sq_sqrt A
+
+@[deprecated CFC.sqrt_mul_sqrt_self (since := "2025-09-22")]
+lemma sqrt_mul_self : CFC.sqrt A * CFC.sqrt A = A := CFC.sqrt_mul_sqrt_self A
+
+lemma eq_of_sq_eq_sq {B : Matrix n n 𝕜} (hB : PosSemidef B) (hAB : A ^ 2 = B ^ 2) : A = B :=
+  CFC.sqrt_sq A ▸ CFC.sqrt_unique (sq B ▸ hAB.symm)
+
+lemma sq_eq_sq_iff {B : Matrix n n 𝕜} (hB : PosSemidef B) : A ^ 2 = B ^ 2 ↔ A = B :=
+  ⟨eq_of_sq_eq_sq hA hB, fun h => h ▸ rfl⟩
+
+@[deprecated CFC.sqrt_sq (since := "2025-09-22")]
+lemma sqrt_sq : CFC.sqrt (A ^ 2) = A := CFC.sqrt_sq A
+
+@[deprecated CFC.sqrt_eq_iff (since := "2025-09-23")]
+lemma eq_sqrt_iff_sq_eq {B : Matrix n n 𝕜} (hB : PosSemidef B) : A = CFC.sqrt B ↔ A ^ 2 = B := by
+  rw [eq_comm, CFC.sqrt_eq_iff B A hB.nonneg hA.nonneg, sq]
+
+@[deprecated CFC.sqrt_eq_iff (since := "2025-09-23")]
+lemma sqrt_eq_iff_eq_sq {B : Matrix n n 𝕜} (hB : PosSemidef B) : CFC.sqrt A = B ↔ A = B ^ 2 := by
+  simpa [eq_comm, sq] using CFC.sqrt_eq_iff A B hA.nonneg hB.nonneg
+
+@[deprecated CFC.sqrt_eq_zero_iff (since := "2025-09-22")]
+lemma sqrt_eq_zero_iff : CFC.sqrt A = 0 ↔ A = 0 := CFC.sqrt_eq_zero_iff A
+
+@[deprecated CFC.sqrt_eq_one_iff (since := "2025-09-23")]
+lemma sqrt_eq_one_iff : CFC.sqrt A = 1 ↔ A = 1 := CFC.sqrt_eq_one_iff A
+
+@[deprecated CFC.isUnit_sqrt_iff (since := "2025-09-22")]
+lemma isUnit_sqrt_iff : IsUnit (CFC.sqrt A) ↔ IsUnit A := CFC.isUnit_sqrt_iff A
+
+lemma inv_sqrt : (CFC.sqrt A)⁻¹ = CFC.sqrt A⁻¹ := by
+  rw [eq_comm, CFC.sqrt_eq_iff _ _  hA.inv.nonneg (CFC.sqrt_nonneg A).posSemidef.inv.nonneg, ← sq,
+    inv_pow', CFC.sq_sqrt A]
+
+end sqrtDeprecated
+
+/-- For `A` positive semidefinite, we have `x⋆ A x = 0` iff `A x = 0`. -/
+theorem dotProduct_mulVec_zero_iff {A : Matrix n n 𝕜} (hA : PosSemidef A) (x : n → 𝕜) :
+    star x ⬝ᵥ A *ᵥ x = 0 ↔ A *ᵥ x = 0 := by
+  classical
+  constructor
+  · obtain ⟨B, rfl⟩ := CStarAlgebra.nonneg_iff_eq_star_mul_self.mp hA.nonneg
+    rw [← Matrix.mulVec_mulVec, dotProduct_mulVec, star_eq_conjTranspose,
+      vecMul_conjTranspose, star_star, dotProduct_star_self_eq_zero]
+    intro h0
+    rw [h0, mulVec_zero]
+  · intro h0
+    rw [h0, dotProduct_zero]
+
+/-- For `A` positive semidefinite, we have `x⋆ A x = 0` iff `A x = 0` (linear maps version). -/
+theorem toLinearMap₂'_zero_iff [DecidableEq n]
+    {A : Matrix n n 𝕜} (hA : PosSemidef A) (x : n → 𝕜) :
+    Matrix.toLinearMap₂' 𝕜 A (star x) x = 0 ↔ A *ᵥ x = 0 := by
+  simpa only [toLinearMap₂'_apply'] using hA.dotProduct_mulVec_zero_iff x
```

Comment:
You can prove the first of these using the faithfulness of the trace, and therefore you can put these in the other file, especially because they don't really involve the *order* (i.e., there is no `≤` appearing).

As a bonus, this will also solve your import headaches.

Reply:
Nevermind, this is much more of a headache than I thought.